addons/web/static/tests/helpers/mock_server.js

```diff
@@ -1031,76 +1040,99 @@ var MockServer = Class.extend({
             case 'name_get':
-                return $.when(this._mockNameGet(args.model, args.args));
+                return this._mockNameGet(args.model, args.args);
 
             case 'name_create':
-                return $.when(this._mockNameCreate(args.model, args.args));
+                return this._mockNameCreate(args.model, args.args);
 
             case 'name_search':
-                return $.when(this._mockNameSearch(args.model, args.args, args.kwargs));
+                return this._mockNameSearch(args.model, args.args, args.kwargs);
 
             case 'onchange':
-                return $.when(this._mockOnchange(args.model, args.args));
+                return this._mockOnchange(args.model, args.args);
 
             case 'read':
-                return $.when(this._mockRead(args.model, args.args, args.kwargs));
+                return this._mockRead(args.model, args.args, args.kwargs);
 
             case 'read_group':
-                return $.when(this._mockReadGroup(args.model, args.kwargs));
+                return this._mockReadGroup(args.model, args.kwargs);
 
             case 'read_progress_bar':
-                return $.when(this._mockReadProgressBar(args.model, args.kwargs));
+                return this._mockReadProgressBar(args.model, args.kwargs);
 
             case 'search_count':
-                return $.when(this._mockSearchCount(args.model, args.args));
+                return this._mockSearchCount(args.model, args.args);
 
             case 'search_read':
-                return $.when(this._mockSearchRead(args.model, args.args, args.kwargs));
+                return this._mockSearchRead(args.model, args.args, args.kwargs);
 
             case 'unlink':
-                return $.when(this._mockUnlink(args.model, args.args));
+                return this._mockUnlink(args.model, args.args);
 
             case 'write':
-                return $.when(this._mockWrite(args.model, args.args));
+                return this._mockWrite(args.model, args.args);
         }
         var model = this.data[args.model];
         if (model && typeof model[args.method] === 'function') {
-            return $.when(this.data[args.model][args.method](args.args, args.kwargs));
+            return this.data[args.model][args.method](args.args, args.kwargs);
         }
 
         throw new Error("Unimplemented route: " + route);
     },
+
+    /**
+     * Dispatch a RPC call to the correct helper function, and add to the request
+     * the abord method like the jQery rpc call.
+     *
+     * @see performRpc
+     * @see _dispatchPerformRpc
+     *
+     * @private
+     * @param {string} route
+     * @param {Object} args
+     * @returns {Deferred<any>}
+     *          Resolved with the result of the RPC. If the RPC should fail, the
+     *          deferred should either be rejected or the call should throw an
+     *          exception (@see performRpc for error handling).
+     */
+    _performRpc: function (route, args) {
```

Comment:
_performRPC is overriden in some addons

Reply:
no problem for that, it's a simplification for not having deferred in deferred for nothing (uses cpu for nothing)